pkgs/development/compilers/riscv-gnu-toolchain/default.nix

```diff
@@ -0,0 +1,125 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, fetchurl
+
+, gcc12
+, glibc
+, newlib
+, gdb
+, qemu
+, musl
+, spike
+, riscv-pk
+
+, autoconf
+, curl
+, automake
+, python3
+, libmpc
+, gawk
+, bison
+, flex
+, texinfo
+, gperf
+, libtool
+, expat
+, gmp
+, mpfr
+, flock
+
+, srcs ? {
+  binutils = {
+    url = "mirror://gnu/binutils/binutils-2.39.tar.bz2";
+    hash = "sha256-2iSoT+8iAQLdJAQt8G/eqFHCYUpTd/hu/6KPM7exYUg=";
+  };
+  dejagnu = {
+    url = "mirror://gnu/dejagnu/dejagnu-1.6.3.tar.gz";
+    hash = "sha256-h9rvrNeVi0pp+IxoVtvRY0JhljxBQHnQw3H1ic1mouM=";
+  };
+  gcc = {
+    url = gcc12.cc.src.url;
+    hash = gcc12.cc.src.outputHash;
+  };
+  glibc = {
+    url = glibc.src.url;
+    hash = glibc.src.outputHash;
+  };
+  newlib = {
+    url = newlib.src.url;
+    hash = "sha256-8pbjcvUTJCJNOHzBFtw3pr05cZh1Z0b5OisC6aXUAVQ=";
```

Comment:
(same for spike and pk) Why override the hash here?

Reply:
The original (newlib/spike/pk).src uses fetchurl, not fetchzip, so its just more convenient

Reply:
However the url could change as the versions are updated, these override would slowly rotten.